File: app/src/main/java/org/codepond/imdemo/service/chat/MessagingServiceModule.java
package org.codepond.imdemo.service.chat;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class MessagingServiceModule {
    @Binds public abstract MessagingService provideMessagingService(FirebaseMessagingService firebaseMessagingService);
}

File: app/src/main/java/org/codepond/imdemo/service/chat/MessagingServiceModule.kt
package org.codepond.imdemo.service.chat

import android.content.Context

import dagger.Module
import dagger.Provides

@Module
class MessagingServiceModule(private val mContext: Context) {

    @Provides internal fun provideMessagingService(): MessagingServiceConnection {
        return XmppMessagingService(mContext)
    }
}
